### ISSUE TITLE
Results for homog.test() seem always incorrect

### ISSUE THREAD

**Issue by Yusuke-Ono** [nicolas-robette/GDAtools#20]
It seems to me that **homog.test()** function always returns incorrect Z-values and p-values.
Could you please check them?

For example, please try the following code. If there are only two groups for comparing,  the results for homogeneity tests must be equal to the ones by typicality tests. But the current **homog.test()** function (in GDATools version 2.0.1) returns different results by **dimtypicality()** function.

```
library(GDAtools)
df <- data.frame(
  x1 = factor(c(1,1,3,1,3,1,2,2,2,2,2)),
  x2 = factor(c(2,3,2,1,1,3,2,2,2,2,1)),
  x3 = factor(c(1,1,1,1,1,1,2,1,2,2,1)),
  x4 = factor(c(1,2,1,3,3,3,2,2,2,2,2)),
  x5 = factor(c(1,1,2,2,2,1,2,1,1,1,1)),
  x6 = factor(c(1,1,2,1,1,1,2,2,2,2,2))
)
resmca <-speMCA(df[, 1:4])
rst.homog.test.x5 <- homog.test(resmca, df$x5, dim=1)
print(rst.homog.test.x5$dim.1$test.stat)
rst.homog.test.x6 <- homog.test(resmca, df$x6, dim=1)
print(rst.homog.test.x6$dim.1$test.stat)

dimtypicality(resmca, df[, c("x5", "x6")], dim = 1)

```

I think that the problem is the following part in **homog.test()** function.

`temp <- abs(outer(vs$coord[,dim[i]],vs$coord[,dim[i]],"-"))/sqrt(eigen[dim[i]])
`
I think the above code should be as below.

`temp <- abs(outer(vs$coord[,dim[i]],vs$coord[,dim[i]],"-"))
`
The **vs$coord[, dim[i]]** has the principal coordinates, which is already equal to the scaled mean points, so you don't need to divide them by the singular values.

**Comment by nicolas-robette** [comment]
You are perfectly right, thanks for sharing the bug ! This should be fixed now.
Bests,
Nicolas

**Comment by Yusuke-Ono** [comment]
Thank you for fixing it, Nocolas-san

I verified the fix in **homog.test()** by using the latest code in the GitHub (on Oct-03-2023 JST). I checked the following three points.

1. Now, the principal coordinates aren't divided by the singular values in the **homog.test()** code.
2. Now, the test R program in my first entry returns the same results as the ones by **dimtypicality()**.
3. Now, the following code returns the almost same result (10.0754) as the one (10.07) on p.86 in Le Roux and Rouanet (2010) _Multiple Correspondence Analysis_, SAGE.  

```
library(readxl)
library(GDAtools)
dt <- read_excel("XXXXXXXXX/Taste_Example.xls")
dt <- subset(dt, Isup == "Active")

dt$Gender_Age <- paste(dt$Gender, "-", dt$Age)
resmca <-speMCA(dt[, 3:6])
rst.homog.test <- homog.test(resmca, dt$Gender_Age, dim=1)
rst.homog.test$dim.1$test.stat[8, 11]
```

Yusuke Ono (Japan)
